src/plots/plots.js

```diff
@@ -2861,6 +2861,31 @@ var sortAxisCategoriesByValueRegex = /(value|sum|min|max) (ascending|descending)
 function sortAxisCategoriesByValue(axList, gd) {
     var affectedTraces = [];
     var i, j, k, l, o;
+
+    function zMapCategory(type, ax, value) {
+        var axLetter = ax._id.charAt(0);
+        if(type === 'histogram2dcontour') {
```

Comment:
Interesting. Could you explain why we need special logic for `histogram2dcontour` but not `histogram2d`?

Reply:
For `histogram2d`: `gd.calcdata[0][0].z` will have dimensions of `x.length` and `y.length`.

For `histogram2dcontour`: `gd.calcdata[0][0].z` will have dimensions of `x.length + 1` and `y.length + 1`. If the `x` or `y` are on categorical axes, the dimensions will be `(x|y).length + 2`. Not sure why that is :man_shrugging:

Reply:
Thanks for the info!